src/services/createRouter.ts

```diff
@@ -227,7 +227,7 @@ export function createRouter<const T extends Routes>(routesOrArrayOfRoutes: T |
     app.provide(routerInjectionKey, router as any)
   }
 
-  const router: Router<T> = {
+  const router: Router<TRoutes> = {
```

Comment:
this is worth noting, the router we create doesn't actually use the `TOptions` argument. In fact if I add it, I'd have to change more about hooks to possibly deviate from just `RegisteredRejectionType`.